qiskit_research/utils/gate_decompositions.py

```diff
@@ -259,3 +261,48 @@ def run(
                 dag.substitute_node_with_dag(node, mini_dag)
 
         return dag
+
+
+class RZXWeylDecomposition(TransformationPass):
+    """
+    Decompose XX, YY, ZZ rotation gates using the Weyl Chamber
+    decomposition, this version accepts Parameter values.
+
+    See https://arxiv.org/abs/2105.01063
+    """
+
+    def run(self, dag: DAGCircuit) -> DAGCircuit:
+        for run in dag.collect_runs(["rxx", "ryy", "rzz"]):
+            for node in run:
+                mini_dag = DAGCircuit()
+                register = QuantumRegister(2)
```

Comment:
Optional: do
```python
a, b = register
```
and then use `a` and `b` instead of `register[0]` and `register[1]` below.

Reply:
Done!